HARK/ConsumptionSaving/tests/test_ConsPortfolioFrameModel.py

```diff
@@ -0,0 +1,146 @@
+import HARK.ConsumptionSaving.ConsPortfolioFrameModel as cpfm
+import numpy as np
+import unittest
+
+
+class PortfolioConsumerTypeTestCase(unittest.TestCase):
+    def setUp(self):
+        # Create portfolio choice consumer type
+        self.pcct = cpfm.PortfolioConsumerFrameType()
+        self.pcct.cycles = 0
+
+        # Solve the model under the given parameters
+
+        self.pcct.solve()
+
+class UnitsPortfolioConsumerTypeTestCase(PortfolioConsumerTypeTestCase):
+    def test_simOnePeriod(self):
+
+        self.pcct.T_sim = 30
+        self.pcct.AgentCount = 10
+        self.pcct.track_vars += ['aNrm']
+        self.pcct.initialize_sim()
+
+        self.assertFalse(
+            np.any(self.pcct.shocks['Adjust'])
+        )
+
+        self.pcct.sim_one_period()
+
+        self.assertAlmostEqual(
+            self.pcct.shocks['PermShk'][0],
+            0.95908471
+        )
+
+        self.assertAlmostEqual(
+            self.pcct.shocks['TranShk'][0],
+            1.03172631
+        )
+
+        self.assertAlmostEqual(
+            self.pcct.shocks['Risky'][0],
+            1.52812838
+        )
+
+        self.assertAlmostEqual(
+            self.pcct.state_now['pLvl'][0],
+            self.pcct.state_prev['pLvl'][0] * self.pcct.shocks['PermShk'][0]
+        )
+
+        self.assertTrue(
+            np.any(self.pcct.shocks['Adjust'][0])
+        )
+
+        self.assertAlmostEqual(
+            self.pcct.state_now["mNrm"][0],
+            self.pcct.state_prev["aNrm"][0] \
+            * self.pcct.Rfree / self.pcct.shocks['PermShk'][0] \
+            + self.pcct.shocks['TranShk'][0] \
+        )
+
+        self.assertAlmostEqual(
+            self.pcct.controls["Share"][0],
+            0.89838257
+        )
+        self.assertAlmostEqual(
+            self.pcct.controls["cNrm"][0],
+            self.pcct.solution[0].cFuncAdj(self.pcct.state_now['mNrm'][0])
+        )
+
+        self.assertAlmostEqual(
+            self.pcct.state_now['aNrm'][0],
+            self.pcct.state_now["mNrm"][0] - self.pcct.controls["cNrm"][0]
+        )
+
+class SimulatePortfolioConsumerTypeTestCase(PortfolioConsumerTypeTestCase):
```

Comment:
Something that would make me confident about the PR would be to have a test where a `ConsPortfolio` and `ConsPortfolioFrame` agent types were created and solved with the default parametrization and their solutions and simulations were compared.

Reply:
That's an interesting idea.
I don't think a comparison between the solutions will be interesting because it's literally just calling the same solution code.
Though I could do a few tests to verify that it's working as expected.

Comparing the simulations would need to take into account a couple things: 
 - differences in sampling. i suppose these could be done by averaging many cases.
 - the fact that in this model the simulation is computing a stochastic transitory shock for the 0th period of life, as opposed to the current default code that for some reason makes this shock 1.0.

Still, a side-by-side plot would be easy to do.